### PR TITLE
[OPIK-3473] [SDK] Python SDK API improvements for log_*_feedback_scores methods

### PR DESCRIPTION
**Pull request by petrotiurin**
## Details

This PR improves the Python SDK API for batch feedback score logging methods (`log_traces_feedback_scores`, `log_spans_feedback_scores`, and `log_threads_feedback_scores`) by introducing a dedicated `BatchFeedbackScoreDict` type that:

- Requires an `id` field for each score (enforced by type checker)
- Supports per-score `project_name` (optional, falls back to method parameter or client default)
- Provides clearer API semantics for batch operations across multiple projects

**Key Changes:**
- Introduced `BatchFeedbackScoreDict` TypedDict with required `id` field and optional per-score `project_name`
- Updated all three `log_*_feedback_scores` methods to use `BatchFeedbackScoreDict` instead of `FeedbackScoreDict`
- Enhanced documentation to clarify that `project_name` parameter serves as a fallback when not specified per-score
- Updated validation helpers to support the new type structure
- Added comprehensive test coverage (345 new test lines for e2e, 259 new test lines for unit tests, 59 new test lines for validator tests)

**Benefits:**
- Type checker now catches missing `id` field in batch operations
- Each score can explicitly specify its project, enabling batch operations across multiple projects
- Clearer API semantics with better documentation
- Backward compatible: `project_name` parameter still works as fallback

## Change checklist
- [ ] User facing
- [x] Documentation update

## Issues
- OPIK-3473

## Testing

Comprehensive test coverage added:
- **E2E tests** (`test_feedback_scores.py`): 345 new lines covering batch operations with per-score project names, fallback behavior, and edge cases
- **Unit tests** (`test_opik_client.py`): 259 new lines testing type validation, error handling, and method behavior
- **Validator tests** (`test_feedback_score_validator.py`): 59 new lines testing feedback score validation logic, including type coercion prevention for `project_name` field
- All existing tests updated to use new `BatchFeedbackScoreDict` type
- Tests verify fallback behavior when `project_name` is not specified per-score

Test scenarios covered:
- Batch logging with per-score project names
- Fallback to method parameter when project_name not specified per-score
- Fallback to client default project when neither specified
- Type validation for required `id` field
- Error handling for invalid score dictionaries
- Validator tests for `project_name` type coercion prevention (rejecting non-string types)

## Documentation

- Updated method docstrings in `opik_client.py` to clarify `project_name` fallback behavior
- Updated Fern documentation for production monitoring and trace annotation
- Added comprehensive type documentation for `BatchFeedbackScoreDict` in `types.py`

### CODE REVIEW

File: sdks/python/src/opik/api_objects/helpers.py [hunk 1]
```diff
@@ -156,8 +156,8 @@ def parse_feedback_score_messages(
     score_messages = [
         parsed_item_class(
             source=constants.FEEDBACK_SCORE_SOURCE_SDK,
-            project_name=project_name,
-            **score_dict,
+            project_name=score_dict.get("project_name") or project_name,
+            **{k: v for k, v in score_dict.items() if k != "project_name"},
         )
         for score_dict in valid_scores
     ]
```
[the remark this answers is not in the document]
Comment (Nimrod007) [reply sdks/python/src/opik/api_objects/helpers.py:167]: @petrotiurin worth a check :) we just discussed something similar

[the remark this answers is not in the document]
Comment (petrotiurin) [reply sdks/python/src/opik/api_objects/helpers.py:167]: Threads can't span multiple projects. Traces, spans and threads are all live on a project level.

File: sdks/python/src/opik/validation/feedback_score.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 
 
 class PydanticWrapper(pydantic.BaseModel):
     model_config = pydantic.ConfigDict(extra="forbid")
-    feedback_score: FeedbackScoreDict
+    feedback_score: BatchFeedbackScoreDict
 
 
-EXPECTED_TYPES = "{'id': str, 'name': str, 'value': float, 'reason': NotRequired[str], 'category_name': NotRequired[str]}"
+EXPECTED_TYPES = "{'id': str, 'name': str, 'value': float, 'reason': NotRequired[str], 'category_name': NotRequired[str], 'project_name': NotRequired[str]}"
```
[the remark this answers is not in the document]
Comment (petrotiurin) [reply sdks/python/src/opik/validation/feedback_score.py:14]: done

**Review by Nimrod007 — APPROVED** [review]
Consider adding this example to the docstring of log_traces_feedback_scores():
something like 
```
# Example: Batch logging across multiple projects
client.log_traces_feedback_scores(
    scores=[
        {"id": trace1_id, "name": "accuracy", "value": 0.95, "project_name": "project-A"},
        {"id": trace2_id, "name": "accuracy", "value": 0.88, "project_name": "project-B"},
        {"id": trace3_id, "name": "accuracy", "value": 0.92},  # Uses project_name parameter or client default
    ],
    project_name="default-project"  # Fallback for scores without explicit project_name
)
```

since we have a complex logic for this

**Review by Nimrod007 — APPROVED** [review]
